Sources/StreamChat/Database/DTOs/UserDTO_Tests.swift

```diff
@@ -141,6 +141,108 @@ class UserDTO_Tests: XCTestCase {
+        // Create explicit userId
+        let userId = UUID().uuidString
+        
+        // Create test payload
+        let payload: UserPayload<NoExtraData> = .dummy(userId: userId)
+        
+        // Save the payload to the db
+        try database.writeSynchronously { session in
+            try session.saveUser(payload: payload)
+        }
+        
+        let changedPayload: ExplicitHashUserPayload = .init(
+            id: userId,
+            name: .unique,
+            imageURL: .unique(),
+            role: .user,
+            createdAt: .unique,
+            updatedAt: .unique,
+            lastActiveAt: .unique,
+            isOnline: !payload.isOnline,
+            isInvisible: !payload.isInvisible,
+            isBanned: !payload.isBanned,
+            extraData: .defaultValue
+        )
+        // Assign it's explicitHast
+        changedPayload.explicitHash = payload.chatHash
+        
+        // Save the changed payload with the same hash
+        try database.writeSynchronously { session in
+            try session.saveUser(payload: changedPayload)
+        }
+        
+        // Load the user from the db and check the fields are correct
+        var loadedUserModel: _ChatUser<NoExtraData>? {
+            database.viewContext.user(id: userId)?.asModel()
+        }
+        
+        // Assert that properties are not changed
+        XCTAssertEqual(payload.name, loadedUserModel?.name)
+        XCTAssertEqual(payload.imageURL, loadedUserModel?.imageURL)
+        XCTAssertEqual(payload.isOnline, loadedUserModel?.isOnline)
+        XCTAssertEqual(payload.isBanned, loadedUserModel?.isBanned)
+        XCTAssertEqual(payload.role, loadedUserModel?.userRole)
+        XCTAssertEqual(payload.createdAt, loadedUserModel?.userCreatedAt)
+        XCTAssertEqual(payload.updatedAt, loadedUserModel?.userUpdatedAt)
+        XCTAssertEqual(payload.lastActiveAt, loadedUserModel?.lastActiveAt)
+        XCTAssertEqual(payload.teams, loadedUserModel?.teams)
+        
+        XCTAssertNotEqual(changedPayload.name, loadedUserModel?.name)
+        XCTAssertNotEqual(changedPayload.imageURL, loadedUserModel?.imageURL)
+        XCTAssertNotEqual(changedPayload.isOnline, loadedUserModel?.isOnline)
+        XCTAssertNotEqual(changedPayload.isBanned, loadedUserModel?.isBanned)
+        XCTAssertNotEqual(changedPayload.role, loadedUserModel?.userRole)
+        XCTAssertNotEqual(changedPayload.createdAt, loadedUserModel?.userCreatedAt)
+        XCTAssertNotEqual(changedPayload.updatedAt, loadedUserModel?.userUpdatedAt)
+        XCTAssertNotEqual(changedPayload.lastActiveAt, loadedUserModel?.lastActiveAt)
+        
+        let newPayload: UserPayload<NoExtraData> = .dummy(userId: userId)
+        
+        // Save the changed payload with the same hash
+        try database.writeSynchronously { session in
+            try session.saveUser(payload: newPayload)
+        }
+        
+        XCTAssertEqual(loadedUserModel?.name, newPayload.name)
```

Comment:
`// Save the changed payload with the same hash` ... and then what? :D Please don't make me think when reading tests. Just explain to me what you're doing in the comments, thanks!

Reply:
fixed in 293c1ee258489fb76a6981f2a809decb044453a0